File: sdk/python/magnatrix_sdk/mesh.py
#!/usr/bin/env python3
"""
mesh.py — MAGNATRIX SDK Mesh Client
Client untuk mesh messaging bus.
"""

import json
import time
import urllib.request
from typing import Dict, List, Optional, Any, Callable


class MeshClient:
    """Client untuk MAGNATRIX mesh messaging."""

    def __init__(self, base_url: str = "http://localhost:8080", api_key: str = ""):
        self.base_url = base_url.rstrip("/")
        self.api_key = api_key

    def send(
        self,
        sender: str,
        msg_type: str,
        payload: Dict[str, Any],
        target: Optional[str] = None,
        priority: int = 5,
    ) -> Dict[str, Any]:
        """Send message ke mesh."""
        # In production, this calls mesh messaging API
        return {
            "status": "sent",
            "sender": sender,
            "target": target,
            "type": msg_type,
            "payload": payload,
            "timestamp": time.time(),
        }

    def broadcast(
        self,
        sender: str,
        msg_type: str,
        payload: Dict[str, Any],
    ) -> Dict[str, Any]:
        """Broadcast message ke semua agent."""
        return self.send(sender, msg_type, payload, target=None)

    def recv(
        self,
        agent_id: str,
        max_items: int = 10,
        msg_type: Optional[str] = None,
    ) -> List[Dict[str, Any]]:
        """Receive messages dari agent inbox."""
        # In production, this polls mesh API
        return []

File: sdk/typescript/src/mesh.ts
/**
 * mesh.ts — MAGNATRIX TypeScript SDK Mesh Client
 */
export class MeshClient {
  private baseUrl: string;

  constructor(baseUrl = "http://localhost:8080") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  async send(sender: string, type: string, payload: any, target?: string, priority = 5): Promise<any> {
    // Placeholder — in production calls mesh API
    return { status: "sent", sender, type, target, payload, timestamp: Date.now() };
  }

  async broadcast(sender: string, type: string, payload: any): Promise<any> {
    return this.send(sender, type, payload);
  }
}
